### PR TITLE
Added  `_ml/agents` and `ml/connectors`  specs 

### PR DESCRIPTION
**Pull request by junweid62**
### Description
This PR documents a subset of the ML Commons APIs consumed by the Flow Framework client.

**Previously covered API specs (in other PRs):**
- `POST  /_plugins/_ml/model_groups/_register`
- `POST  /_plugins/_ml/models/_register`
- `DELETE  /_plugins/_ml/models/{model_id}`
- `POST /_plugins/_ml/models/{model_id}/_deploy`
- `POST /_plugins/_ml/models/{model_id}/_undeploy`

**New API specs added in this PR:**
- `POST  /_plugins/_ml/agents/_register`
- `DELETE  /_plugins/_ml/agents/{agent_id}`
- `POST  /_plugins/_ml/connectors/_create`
- `DELETE  /_plugins/_ml/connectors/{connector_id}`

### Issues Resolved
Part of #228 
Task 3 of [#833](https://github.com/opensearch-project/flow-framework/issues/833)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/schemas/ml._common.yaml [hunk 1]
```diff
@@ -156,3 +156,131 @@ components:
...
+          items:
+            $ref: '#/components/schemas/Action'
+      required:
+        - actions
+        - credential
+        - description
+        - name
+        - parameters
+        - protocol
+        - version
```
Comment (dbwiddis) [inline spec/schemas/ml._common.yaml:187]: This is exactly the data we're going to want to pull programmatically to validate our templates.  Glad to see it being added!

**Review by dbwiddis — APPROVED** [review]
LGTM from an API standpoint!

File: CHANGELOG.md [hunk 2]
```diff
@@ -95,7 +95,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added `/_plugins/_flow_framework/`, `_search`, `state/_search`, `_provision`, `_deprovision`, `_steps`, and `_status` ([#508](https://github.com/opensearch-project/opensearch-api-specification/issues/508)) ([#833](https://github.com/opensearch-project/flow-framework/issues/833))
 - Added `/_plugins/_ism/policies`, `add`, `remove`, `change_policy`, `explain`, and `retry` ([#568](https://github.com/opensearch-project/opensearch-api-specification/pull/568)) ([#578](https://github.com/opensearch-project/opensearch-api-specification/pull/578))
 - Added `/_plugins/refresh_search_analyzers` ([#578](https://github.com/opensearch-project/opensearch-api-specification/pull/578))
-
```
Comment (dblock) [inline CHANGELOG.md:98]: Re-add this empty line below.

File: CHANGELOG.md [hunk 3]
```diff
@@ -95,7 +95,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
...
+- Added `/_plugins/_ml/agents/_register`, `/_plugins/_ml/connectors/_create`, `DELETE /_plugins/_ml/agents/{agent_id}`, `DELETE /_plugins/_ml/connectors/{connector_id}`([#228](https://github.com/opensearch-project/opensearch-api-specification/issues/228)
```
Comment (dblock) [inline CHANGELOG.md:98]:
```suggestion
- Added `/_plugins/_ml/agents/_register`, `/_plugins/_ml/connectors/_create`, `DELETE /_plugins/_ml/agents/{agent_id}`, `DELETE /_plugins/_ml/connectors/{connector_id}` ([#228](https://github.com/opensearch-project/opensearch-api-specification/issues/228))
```

**Review by dblock — CHANGES_REQUESTED** [review]
LGTM, small stuff for the broken link in CHANGELOG (missing a `)`).

**Review by dblock — APPROVED** [review]

**Pull request merged**